src/conn/conn_capacity.c

```diff
@@ -356,9 +356,16 @@ __wt_capacity_throttle(WT_SESSION_IMPL *session, uint64_t bytes, WT_THROTTLE_TYP
 
     conn = S2C(session);
     cap = &conn->capacity;
-    /* NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores) */
+
     capacity = steal_capacity = 0;
     reservation = steal = NULL;
+
+    /*
+     * The assignment to capacity is a dead assignment but gcc complains it may be used
+     * uninitialized. Quiet clang static analyzer by casting to void.
+     */
+    WT_DEAD_STORE(capacity);
```

Comment:
Suggestion: When `ENABLE_STRICT=1` we set `-Werror=switch` and both clang and gcc will catch missing cases in switch statements. Since `ENABLE_STRICT` defaults to on I'm ok with removing both `capacity = 0` and the `WT_DEAD_STORE` macro, instead trusting the compiler to detect missing switch case.

Reply:
Gcc doesn't compile if you remove capacity =  0. Which is why I kept it.